### DISCUSSION TITLE
Baseball Draft Info

### DISCUSSION THREAD

**Discussion by K-Roberts** [cwendt94/espn-api#528] [General] [answered]
Let me start off by saying this espn-api project is amazing!  I have been using it for years to help me maintain my football league.  I am trying to do a similar thing for my baseball league.  I'm curious if it would be possible to add espn draft information to the baseball League()?  I would like to be able to siphon through the espn draft and know what position player X was taken at (ex. 18th pos). Thanks in advance, you all are awesome!

**Comment by cwendt94** [comment]
Yeah that should be an easy add! Let me take a look at that!

**Reply by K-Roberts** [reply]
Thank you, Christian!

**Reply by cwendt94** [reply, marked as answer]
Draft is now is version v0.35.0!

**Reply by K-Roberts** [reply]
@cwendt94 - thank you for updating this - the work you've done on this is amazing!  

What I do with this info is create a script that writes to csv the teams and each players respective draft spot (for a keeper league) to help me manage the league better.

v0.35.0 works for football after shifting Draft to Base_League, but I am having trouble accessing the draft in baseball still.

Here is my script - if you have interest/availability, are you able to spot anything wrong when I'm trying to get the draft info?  This is an identical script to the football one that works.

Thank you.

///////////////////////////////////
import csv
import math
from espn_api.baseball import League

#league home page
league = League(league_id=26830, year=2023)

#separating teams
t1 = league.teams[0]
t2 = league.teams[1]
t3 = league.teams[2]
t4 = league.teams[3]
t5 = league.teams[4]
t6 = league.teams[5]
t7 = league.teams[6]
t8 = league.teams[7]
t9 = league.teams[8]
t10 = league.teams[9]

teams = [t1,t2,t3,t4,t5,t6,t7,t8,t9,t10]

#team names
teamnames = []
for i in teams:
	teamnames.append(i.team_name)

#team rosters
trueteams = []
for i in teams:
	trueteams.append(i.team_name)
	for j in i.roster:
		trueteams.append(j.name)
	trueteams.append(' ')


#draft order of players
draft = []
for i in range(len(league.draft)):
	draft.append(league.draft[i].playerName)


#write to csv	
with open('FBaseball_Teams.csv', 'w') as writeFile:
	writer = csv.writer(writeFile, delimiter = '\n')
	for i in trueteams:
		if i in draft:
			col = [i, str(math.ceil((draft.index(i) + 1)/10))]
			writer.writerow([', '.join(str(x) for x in col)])
		elif i not in teamnames and i != ' ':
			col = [i, 'Free Agent']
			writer.writerow([', '.join(str(x) for x in col)])
		else:
		        writer.writerow([i])
writeFile.close()
///////////////////////////////////

**Reply by cwendt94** [reply]
What issues are you seeing when trying it with baseball?

**Reply by K-Roberts** [reply]
@cwendt94 - I cannot get the draft information with the same calls as I can for Football.  I will keep trying to troubleshoot why.

For example, league.draft is unpopulated in baseball, yet it's populated in football.  

Thanks again for all you do - this project is really amazing
